In [1]:
import os
import regex as re

try:
    import fitz  # this is pymupdf
except:
    %pip install fitz
import requests

In [2]:
with fitz.open("pml1-listoffigures.pdf") as doc:
    text = ""
    for page in doc:
        text += page.get_text()

# print(text)

In [3]:
pattern = r"[\n][0-9]+[.][0-9]+[\n]"
fig_numbers = re.findall(pattern, text)
len(fig_numbers)

431

In [4]:
captions = re.compile(pattern).split(text)[1:]
len(captions)

431

In [6]:
seperator_pattern = r"[\s\n]"
scripts = {}
book_no = 1
for no, cap in zip(fig_numbers, captions):
    words = re.compile(seperator_pattern).split(cap)
    fig_no = no.strip()
    chapter_no = int(fig_no.split(".")[0])
    base_url_ipynb = f"https://github.com/probml/pyprobml/blob/master/notebooks/book{book_no}/{chapter_no:02d}"
    script_names = []

    for each in words:
        if ".py" in each:
            script_name = each[: each.find(".py")]
            script_names.append(os.path.join(base_url_ipynb, script_name + ".ipynb"))
        elif ".ipynb" in each:
            script_name = each[: each.find(".ipynb")]
            script_names.append(os.path.join(base_url_ipynb, script_name + ".ipynb"))
    print(fig_no, script_names)
    scripts[fig_no] = script_names

1.1 []
1.2 []
1.3 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/iris_plot.ipynb']
1.4 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/iris_dtree.ipynb']
1.5 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/linreg_residuals_plot.ipynb']
1.6 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/linreg_2d_surface_demo.ipynb']
1.7 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/linreg_poly_vs_degree.ipynb']
1.8 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/iris_kmeans.ipynb']
1.9 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/iris_pca.ipynb']
1.10 []
1.11 []
1.12 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/mnist_viz_tf.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/emnist_viz_jax.ipynb']
1.13 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/fashion_viz_tf.ipynb', 'https://github.com/p

In [7]:
more_than_one = 0
for each in scripts:
    if len(scripts[each]) > 1:
        print(each, scripts[each])
        more_than_one += 1
more_than_one

1.12 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/mnist_viz_tf.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/emnist_viz_jax.ipynb']
1.13 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/fashion_viz_tf.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/cifar_viz_tf.ipynb']
2.2 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/02/gauss_plot.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/02/quantile_plot.ipynb']
2.17 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/02/beta_dist_plot.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/02/gamma_dist_plot.ipynb']
4.14 ['https://github.com/probml/pyprobml/blob/master/notebooks/book1/04/dirichlet_3d_triangle_plot.ipynb', 'https://github.com/probml/pyprobml/blob/master/notebooks/book1/04/dirichlet_3d_spiky_plot.ipynb']
4.20 ['https://github.com/probml/pyprobml/blob/master/note

25

In [8]:
cnt = 0
for fig in scripts:
    if scripts[fig] != []:
        cnt += 1
cnt

219

In [59]:
dead_urls = []
for fig in scripts:
    for url in scripts[fig]:
        resp = requests.get(url)
        if resp.status_code != 200:
            print(fig, url)
            dead_urls.append([fig, url])

1.12 https://github.com/probml/pyprobml/blob/master/notebooks/book1/01/emnist_viz_jax.ipynb
3.13 https://github.com/probml/pyprobml/blob/master/notebooks/book1/03/mix_ber_em_mnist.ipynb
4.17 https://github.com/probml/pyprobml/blob/master/notebooks/book1/04/gaussInferParamsMean2d.ipynb
9.5 https://github.com/probml/pyprobml/blob/master/notebooks/book1/09/fisher_vowel_demo.ipynb
9.6 https://github.com/probml/pyprobml/blob/master/notebooks/book1/09/naive_bayes_mnist_jax.ipynb
9.7 https://github.com/probml/pyprobml/blob/master/notebooks/book1/09/naive_bayes_mnist_jax.ipynb


KeyboardInterrupt: 

1.12: Delete emnist_viz_pytorch.ipynb from pyprobml and add emnist_viz_jax.ipynb from probml-notebooks/notebooks
3.13: Rename mix_ber_em_mnist.ipynb to mix_bernoulli_em_mnist.ipynb in this script
4.17: Rename gaussInferParamsMean2d.ipynb to gauss_infer_2d.ipynb
9.5: Rename fisher_vowel_demo.ipynb to fisher_discrim_vowel.ipynb
9.6, 9.7: Delete naive_bayes_mnist.ipynb from pyprobml and Move naive_bayes_mnist_jax.ipynb from probml-notebooks/notebooks-d2l to pyprobml
15.25: Delete positional_encoding.ipynb from pyprobml and Move positional_encoding_jax.ipynb from probml-notebooks/notebooks-d2l to pyprobml 
19.14: Move hbayes_maml.ipynb from probml-notebooks/notebooks to pyprobml
20.24, 20.25, 20.26: Move ae_mnist_conv.ipynb from probml-notebooks/notebooks to pyprobml